contracts/Quest1155.sol

```diff
@@ -121,6 +126,50 @@ contract Quest1155 is ERC1155Holder, ReentrancyGuardUpgradeable, PausableUpgrade
         emit Queued(block.timestamp);
     }
 
+    /// @dev recover the signer from a hash and signature
+    /// @param hash_ The hash of the message
+    /// @param signature_ The signature of the hash
+    function recoverSigner(bytes32 hash_, bytes memory signature_) public view returns (address) {
+        return ECDSA.recover(ECDSA.toEthSignedMessageHash(hash_), signature_);
+    }
+
+    /// @dev universal dynamic claim function
+    /// @param signature_ The signature of the data
+    /// @param data_ The data to decode for the claim
+    function claim(bytes calldata signature_, bytes calldata data_) external payable
+        // nonReentrant need to be sure we don't need this
+        whenNotEnded
```

Comment:
Is there a way to setup the quest creators withdraw function to remove the need for the whenNotEnded for the claim?

Reply:
I'm aligned with your thinking - not worth it.